game-core/src/main/java/games/strategy/triplea/delegate/EndTurnDelegate.java

```diff
@@ -206,51 +195,73 @@ private String addUnitCreatedResources(final IDelegateBridge bridge) {
     return endTurnReport.toString();
   }
 
+  /**
+   * Find all of the resources that will be created by units on the map.
+   */
+  public static IntegerMap<Resource> findUnitCreatedResources(final PlayerID player, final GameData data) {
+    final IntegerMap<Resource> resourceTotalsMap = new IntegerMap<>();
+    final Predicate<Unit> myCreatorsMatch = Matches.unitIsOwnedBy(player).and(Matches.unitCreatesResources());
+    for (final Territory t : data.getMap().getTerritories()) {
+      final Collection<Unit> myCreators = CollectionUtils.getMatches(t.getUnits().getUnits(), myCreatorsMatch);
+      for (final Unit unit : myCreators) {
+        final IntegerMap<Resource> generatedResourcesMap = UnitAttachment.get(unit.getType()).getCreatesResourcesList();
+        resourceTotalsMap.add(generatedResourcesMap);
+      }
+    }
+    for (final Resource resource : resourceTotalsMap.keySet()) {
+      if (resource.getName().equals(Constants.PUS)) {
+        resourceTotalsMap.put(resource, resourceTotalsMap.getInt(resource) * Properties.getPuMultiplier(data));
+      }
+    }
+    return resourceTotalsMap;
+  }
+
+  /**
+   * Find the PU value of national objectives that are currently met.
+   */
+  public static int findNationalObjectivePus(final PlayerID player, final GameData data, final IDelegateBridge bridge) {
+    final List<RulesAttachment> objectives = new ArrayList<>();
+    final HashMap<ICondition, Boolean> testedConditions =
+        testNationalObjectivesAndTriggers(player, data, bridge, new HashSet<>(), objectives);
+    int pus = 0;
+    for (final RulesAttachment rule : objectives) {
+      final int uses = rule.getUses();
+      if (uses == 0 || !rule.isSatisfied(testedConditions)) {
+        continue;
+      }
+      int toAdd = rule.getObjectiveValue();
+      toAdd *= Properties.getPuMultiplier(data);
+      toAdd *= rule.getEachMultiple();
```

Comment:
Can these three statements be collapsed to the single statement:

```java
final int toAdd = rule.getObjectiveValue()
    * Properties.getPuMultiplier(data)
    * rule.getEachMultiple();
```

Or something like that?

Reply:
Updated.